tests/ui/never_type/diverging-place-match.rs

```diff
@@ -0,0 +1,13 @@
+#![feature(never_type)]
+
+fn make_up_a_value<T>() -> T {
+    unsafe {
+    //~^ ERROR mismatched types
+        let x: *const ! = 0 as _;
+        let _: ! = *x;
```

Comment:
It's probably on your radar, but it would be good to have more tests:
- also test `let _` without type annotations (that has sometimes behavior differently)
- also test `match *x { _ => {} }`
- also test the case where a field projection is involved, not just a deref: `x: *const (!, ())`, and then do `(*x).0`.

Reply:
Thanks. I appreciate making sure this is thorough -- I'll have to rewrite this history and clean up the PR, but I'll be certain to include the tests.